### PR TITLE
README improvements

### PR DESCRIPTION
**Pull request by myronmarston**
My main goal here was to add a section on scope.  After reading http://therealadam.com/2015/03/01/its-not-your-fault-if-your-tools-confuse-you/ I realized we don't document scope very well.  While reading through what we had in the README I noticed a number of things that could be improved so I went ahead and addressed those.

### CODE REVIEW

**Comment by therealadam** [comment]
The prose on scoping is great. The scoping example code was initially surprising to me due to calling a method on `super`, which is not a thing I usually see or expect to see. The example transliterating the DSL to a hand-coded Ruby class is great too.

**Comment by myronmarston** [comment]
> The prose on scoping is great. The scoping example code was initially surprising to me due to calling a method on super, which is not a thing I usually see or expect to see. The example transliterating the DSL to a hand-coded Ruby class is great too.

Thanks.  I adapted the example from a prior response I had sent someone who had specifically asked if you can use `super` so it was a very arbitrary example that was attempting to demonstrate that...but it's totally unneeded here.  I pushed an update...do you like that better?

**Comment by myronmarston** [comment]
I also added a bit about how the examples would get run.

**Comment by therealadam** [comment]
Yep the code is exactly what I'd expect now. 

--
~akk
http://therealadam.com

File: README.md [hunk 1]
```diff
@@ -162,26 +167,106 @@ RSpec.describe Hash do
...
 directory, and the `rspec` command will read them as though you typed them on
 the command line.
 
-## autotest integration
-
-rspec-core no longer ships with an Autotest extension, if you require Autotest
-integration, please use the `rspec-autotest` gem and see [rspec/rspec-autotest](https://github.com/rspec/rspec-autotest)
-for details
-
-## get started
```
Comment (JonRowe) [inline README.md]: Oops ;)

**Comment by JonRowe** [comment]
This LGTM :) nice improvement, although I couldn't read the blog post that inspired it (blank page)

**Pull request merged**